secure_message/repository/modifier.py

```diff
@@ -133,7 +133,58 @@ def _mark_read(message, user):
         Modifier.remove_label(unread, message, user)
 
     @staticmethod
-    def close_conversation(metadata, user):
+    def patch_conversation(request_data: dict, conversation: Conversation):
+        """
+        Goes through each key of the request and updates the conversation object with it if it doesn't match
+        what is in the conversation.  Won't update `is_closed`, `closed_by` and `closed_by_uuid` as these are currently
+        handled in a different manner in another function.
+
+        :param request_data: Json containing which fields should be patched
+        :param conversation: An object containing the currently saved database data
+        """
+        bound_logger = logger.bind(conversation_id=conversation.id)
+
+        for field in ['category']:
+            # Looks a bit awkward but getattr/setattr lets us loop over all the fields as we can't access fields
+            # in the object like a dictionary.
+
+            if request_data.get(field) and request_data.get(field) != getattr(conversation, field):
+                setattr(conversation, field, request_data[field])
+            else:
+                logger.info("Value is the same as what is in the database, leaving unchanged", field=field)
+        try:
+            db.session.commit()
+        except SQLAlchemyError:
+            db.session.rollback()
+            bound_logger.exception("Database error occurred while opening conversation")
```

Comment:
Are we swallowing the exception here? I'd rather we had details of the exception in the logs

Reply:
logger.exception puts the stack trace in the logs for free